File: articles/experimental-moral-psychology-and-the-study-of-ethical-intuition/c/intuition_judgment_score.c
/*
Synthetic moral judgment score example for Experimental Moral Psychology.
Educational scaffold only.

Compile:
  cc c/intuition_judgment_score.c -o outputs/tables/intuition_judgment_score_c
Run:
  ./outputs/tables/intuition_judgment_score_c
*/

#include <stdio.h>

double moral_judgment_score(
    double intuition,
    double reflection,
    double norm_sensitivity,
    double consequence_sensitivity
) {
    return 0.30 * intuition
         + 0.20 * reflection
         + 0.25 * norm_sensitivity
         + 0.25 * consequence_sensitivity;
}

int main(void) {
    double score = moral_judgment_score(0.8, 0.4, 0.7, 0.6);
    printf("Synthetic moral judgment score: %.3f\n", score);
    return 0;
}

File: articles/experimental-moral-psychology-and-the-study-of-ethical-intuition/rust/intuition_judgment_score.rs
// Synthetic moral judgment score example for Experimental Moral Psychology.
// Educational scaffold only.
//
// Compile:
//   rustc rust/intuition_judgment_score.rs -o outputs/tables/intuition_judgment_score_rust
// Run:
//   ./outputs/tables/intuition_judgment_score_rust

fn main() {
    let indicators = [
        ("intuition", 0.8_f64, 0.30_f64),
        ("reflection", 0.4, 0.20),
        ("norm_sensitivity", 0.7, 0.25),
        ("consequence_sensitivity", 0.6, 0.25),
    ];

    let score: f64 = indicators
        .iter()
        .map(|(_, value, weight)| value * weight)
        .sum();

    println!("Synthetic moral judgment score: {:.3}", score);
}
